db-server/test/backend/db_tests.js

```diff
@@ -992,38 +1008,59 @@ module.exports = function (config, DB) {
           })
       })
 
-      it('should fail to update a device with unknown capabilities', () => {
+      it('availableCommands are not cleared if not specified', () => {
+        const newDevice = Object.assign({}, deviceInfo)
+        delete newDevice.availableCommands
+        return db.updateDevice(accountData.uid, deviceInfo.deviceId, newDevice)
+          .then(() => {
+            return db.device(accountData.uid, deviceInfo.deviceId)
+          })
+          .then(device => assert.deepEqual(device.availableCommands, {
+            'https://identity.mozilla.com/cmd/display-uri': 'metadata-bundle'
+          }))
+      })
+
+      it('availableCommands are overwritten on update', () => {
         const newDevice = Object.assign({}, deviceInfo, {
-          capabilities: ['unknown', 'newmessages']
+          availableCommands: {
+            foo: 'bar',
+            second: 'command'
+          }
         })
         return db.updateDevice(accountData.uid, deviceInfo.deviceId, newDevice)
-          .then(assert.fail, (err) => {
-            assert.equal(err.code, 400, 'err.code')
-            assert.equal(err.errno, 139, 'err.errno')
-            return db.accountDevices(accountData.uid)
+          .then(() => {
+            return db.device(accountData.uid, deviceInfo.deviceId)
           })
-          .then((devices) => assert.deepEqual(devices[0].capabilities, ['messages']))
+          .then(device => assert.deepEqual(device.availableCommands, {
+            foo: 'bar',
+            second: 'command'
+          }))
       })
 
-      it('capabilities are not cleared if not specified', () => {
-        const newDevice = Object.assign({}, deviceInfo)
-        delete newDevice.capabilities
+      it('availableCommands can be overwritten with an empty map', () => {
+        const newDevice = Object.assign({}, deviceInfo, {
+          availableCommands: {}
+        })
         return db.updateDevice(accountData.uid, deviceInfo.deviceId, newDevice)
           .then(() => {
-            return db.accountDevices(accountData.uid)
+            return db.device(accountData.uid, deviceInfo.deviceId)
           })
-          .then((devices) => assert.deepEqual(devices[0].capabilities, ['messages']))
+          .then(device => assert.deepEqual(device.availableCommands, {}))
```

Comment:
Does this just duplicate the assertion on line `977`, or is there a difference between them?

Reply:
You're right, it's a dupe, I'll remove it.